x-pack/otel/exporter/logstashexporter/internal/event_test.go

```diff
@@ -0,0 +1,482 @@
+					Metadata: client.NewMetadata(map[string][]string{
+						"beat_name":    {"filebeat"},
+						"beat_version": {"8.0.0"},
+					}),
+				}
+				return client.NewContext(ctx, info)
+			},
+			expected: map[string]any{
+				"beat":    "filebeat",
+				"version": "8.0.0",
+			},
+		},
+		{
+			name: "missing beat name",
+			setupCtx: func() context.Context {
+				ctx := context.Background()
+				info := client.Info{
+					Metadata: client.NewMetadata(map[string][]string{
+						"beat_version": {"8.0.0"},
+					}),
+				}
+				return client.NewContext(ctx, info)
+			},
+			expected: map[string]any{
+				"beat":    "",
+				"version": "8.0.0",
+			},
+		},
+		{
+			name: "missing beat version",
+			setupCtx: func() context.Context {
+				ctx := context.Background()
+				info := client.Info{
+					Metadata: client.NewMetadata(map[string][]string{
+						"beat_name": {"filebeat"},
+					}),
+				}
+				return client.NewContext(ctx, info)
+			},
+			expected: map[string]any{
+				"beat":    "filebeat",
+				"version": "",
+			},
+		},
+		{
+			name: "no metadata",
+			setupCtx: func() context.Context {
+				ctx := context.Background()
+				info := client.Info{
+					Metadata: client.NewMetadata(map[string][]string{}),
+				}
+				return client.NewContext(ctx, info)
+			},
+			expected: map[string]any{
+				"beat":    "",
+				"version": "",
+			},
+		},
+		{
+			name: "no client info in context",
+			setupCtx: func() context.Context {
+				return context.Background()
+			},
+			expected: map[string]any{
+				"beat":    "",
+				"version": "",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := tt.setupCtx()
+
+			metadata := parseEventMetadata(ctx)
+
+			assert.Equal(t, tt.expected, metadata)
+		})
+	}
+}
+
+func TestIsBeatsEvent(t *testing.T) {
+	tests := []struct {
+		name     string
+		metadata map[string]any
+		expected bool
+	}{
+		{
+			name: "valid beats event",
+			metadata: map[string]any{
+				"beat":    "filebeat",
+				"version": "8.0.0",
+			},
+			expected: true,
+		},
+		{
+			name: "missing beat field",
+			metadata: map[string]any{
+				"version": "8.0.0",
+			},
+			expected: false,
+		},
+		{
+			name: "nil beat field",
+			metadata: map[string]any{
+				"beat":    nil,
+				"version": "8.0.0",
+			},
+			expected: false,
+		},
+		{
+			name: "empty beat field",
+			metadata: map[string]any{
+				"beat":    "",
+				"version": "8.0.0",
+			},
+			expected: false,
+		},
+		{
+			name:     "empty metadata",
+			metadata: map[string]any{},
+			expected: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := isBeatsEvent(tt.metadata)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func TestGetBeatVersion(t *testing.T) {
+	tests := []struct {
+		name     string
+		setupCtx func() context.Context
+		expected string
+	}{
+		{
+			name: "version exists",
+			setupCtx: func() context.Context {
+				ctx := context.Background()
```

Comment:
Please double-check all usages of `context`, most of them can likely be replaced with `t.Context`.